docs/guide/en/schema/typecasting.md

```diff
@@ -0,0 +1,281 @@
+# Type casting values
+
+Type casting is the process of converting a value from one data type to another. In the context of the database,
+type casting is used to ensure that values are saved and retrieved in the correct type.
+
+## Casting values to be saved in the database
+
+When saving a value to the database, the value must be in the correct type. For example, if saving a value to a column
+that is of type `integer`, the value must be an integer.
+
+To ensure that the value is saved in the correct type, `ColumnInterface::dbTypecast()` method can be used to cast 
+the value to the correct type. Some methods in the DB library, such as `CommandInterface::insert()`, automatically
+cast values to the correct type.
+
+```php
+use Yiisoft\Db\Connection\ConnectionInterface;
+
+/** @var ConnectionInterface $db */
+$command = $db->createCommand();
+$command->insert('customer', [
+    'name' => 'John Doe',
+    'is_active' => true,
+]);
+$command->execute();
+```
+
+In the example above, the value of `is_active` is a `boolean`, but the column `is_active` can be of type `bit`.
+The `CommandInterface::insert()` method will automatically cast the value to the correct type.
+
+## Casting values retrieved from the database
+
+When you retrieve a value from the database, the value can be returned in a different type than you expect.
+For example, a value that is stored as an `float` in the database can be returned as a `string`. This is because 
+the database driver do not convert some data types when retrieves values.
+
+To ensure that the value is returned in the correct type, you can use `ColumnInterface::phpTypecast()` method to cast 
+the value to the correct type.
+
+```php
+use Yiisoft\Db\Connection\ConnectionInterface;
+
+/** @var ConnectionInterface $db */
+$command = $db->createCommand('SELECT * FROM {{customer}} WHERE id = 1');
+
+$row = $command->queryOne();
+$isActive = $row['is_active'];
+
+// Cast the value to the correct type
+$isActive = $db->getTableSchema('customer')->getColumn('is_active')->phpTypecast($isActive);
```

Comment:
Is there a way to cast the whole row at once?

Reply:
No, but there is a task about how to do it automatically
https://github.com/yiisoft/db/issues/771